src/components/views/dialogs/DMInviteDialog.js

```diff
@@ -109,6 +108,59 @@ export default class DMInviteDialog extends React.PureComponent {
         return recents;
     }
 
+    _buildSuggestions(): {userId: string, user: RoomMember} {
+        const maxConsideredMembers = 200;
+        const client = MatrixClientPeg.get();
+        const excludedUserIds = [client.getUserId(), SdkConfig.get()['welcomeUserId']];
+        const joinedRooms = client.getRooms()
+            .filter(r => r.getMyMembership() === 'join')
+            .filter(r => r.getJoinedMemberCount() <= maxConsideredMembers);
+        const memberRooms = joinedRooms.reduce((members, room) => {
+            const joinedMembers = room.getJoinedMembers().filter(u => !excludedUserIds.includes(u.userId));
+            for (const member of joinedMembers) {
+                if (!members[member.userId]) {
+                    members[member.userId] = {
+                        member: member,
+                        // Track the room size of the 'picked' member so we can use the profile of
+                        // the smallest room (likely a DM).
+                        pickedMemberRoomSize: room.getJoinedMemberCount(),
+                        rooms: [],
+                    };
+                }
+
+                members[member.userId].rooms.push(room);
+
+                if (room.getJoinedMemberCount() < members[member.userId].pickedMemberRoomSize) {
+                    members[member.userId].member = member;
+                    members[member.userId].pickedMemberRoomSize = room.getJoinedMemberCount();
+                }
+            }
+            return members;
+        }, {/* userId => {member, rooms[]} */});
```

Comment:
What are these commented out bits? Should they be removed...?

Reply:
Documentation for what this thing is generating, likewise below.

Reply:
Hmm, okay, I guess I am just not familiar with this style of documenting, is it a TypeScript style or what?

Reply:
it's probably carryover from my ancient Java days. It's trying to say the key is the user ID and the value is an object containing a member and array of rooms (using the property names instead of types).

If it's not adding value, it can be removed.

Reply:
It does seem useful to have recorded. I think I am mostly reacting with surprise because of:

* the placement inside the reduce initial value
* the use of `=>` even though it's not a function

Maybe it would be less surprising above the variable and without an arrow, as in:

```js
// { userId: {member, rooms[]} }
const memberRooms = joinedRooms.reduce((members, room) => {
```